src/components/TextInput/BaseTextInput/index.native.tsx

```diff
@@ -21,37 +21,71 @@ import useStyleUtils from '@styles/useStyleUtils';
 import useThemeStyles from '@styles/useThemeStyles';
 import variables from '@styles/variables';
 import CONST from '@src/CONST';
-import * as baseTextInputPropTypes from './baseTextInputPropTypes';
-
-function BaseTextInput(props) {
+import type BaseTextInputProps from './types';
+import type {BaseTextInputRef} from './types';
+
+function BaseTextInput(
+    {
+        label = '',
+        value,
+        defaultValue,
+        placeholder = '',
+        errorText = '',
+        icon = null,
+        textInputContainerStyles,
+        containerStyles,
+        inputStyle,
+        forceActiveLabel = false,
+        autoFocus = false,
+        disableKeyboard = false,
+        autoGrow = false,
+        autoGrowHeight = false,
+        hideFocusedState = false,
+        maxLength,
+        hint = '',
+        onInputChange = () => {},
+        shouldDelayFocus = false,
+        submitOnEnter = false,
+        multiline = false,
+        shouldInterceptSwipe = false,
+        autoCorrect = true,
+        prefixCharacter,
+        inputID,
+        ...inputProps
+    }: BaseTextInputProps,
+    ref: BaseTextInputRef,
+) {
     const theme = useTheme();
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
-    const initialValue = props.value || props.defaultValue || '';
-    const initialActiveLabel = props.forceActiveLabel || initialValue.length > 0 || Boolean(props.prefixCharacter);
-    const isMultiline = props.multiline || props.autoGrowHeight;
+    const {translate} = useLocalize();
+    const {hasError = false} = inputProps;
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    const initialValue = value || defaultValue || '';
+    const initialActiveLabel = !!forceActiveLabel || initialValue.length > 0 || !!prefixCharacter;
+    const isMultiline = multiline || autoGrowHeight;
 
     const [isFocused, setIsFocused] = useState(false);
-    const [passwordHidden, setPasswordHidden] = useState(props.secureTextEntry);
+    const [passwordHidden, setPasswordHidden] = useState(inputProps.secureTextEntry);
     const [textInputWidth, setTextInputWidth] = useState(0);
     const [textInputHeight, setTextInputHeight] = useState(0);
-    const [height, setHeight] = useState(variables.componentSizeLarge);
-    const [width, setWidth] = useState();
+    const [height, setHeight] = useState<number>(variables.componentSizeLarge);
+    const [width, setWidth] = useState<number | null>(null);
     const labelScale = useRef(new Animated.Value(initialActiveLabel ? styleConst.ACTIVE_LABEL_SCALE : styleConst.INACTIVE_LABEL_SCALE)).current;
     const labelTranslateY = useRef(new Animated.Value(initialActiveLabel ? styleConst.ACTIVE_LABEL_TRANSLATE_Y : styleConst.INACTIVE_LABEL_TRANSLATE_Y)).current;
 
-    const input = useRef(null);
+    const input = useRef<TextInput>(null);
```

Comment:
```suggestion
    const input = useRef<TextInput & HTMLElement>(null);
```

Reply:
Just out of curosity, if this is the native only file, will it ever be an HTMLElement? @VickyStash

Reply:
@kubabutkiewicz Good question, I guess we can make it just `TextInput` if we update `isInputAutoFilled` in the way like
```
export default function isInputAutoFilled(input: TextInput | HTMLElement | null): boolean {
    if ((input && !('matches' in input)) || !input?.matches) {
        return false;
    }
    if (isSelectorSupported(':autofill')) {
        return input.matches(':-webkit-autofill') || input.matches(':autofill');
    }
    return input.matches(':-webkit-autofill');
}
```
Feel free to improve the first if expression 😅